File: Data Structures & Algorithms/combination-target-sum/submission-0.java
class Solution {
    List<List<Integer>> result = new ArrayList<>();
    int size;
    int k;
    public List<List<Integer>> combinationSum(int[] nums, int target) {
        size = nums.length;
        k = target;
        backtrack(nums, 0, new ArrayList<>(), 0);
        return result;
    }

    void backtrack(int[] nums,int start, List<Integer> subset, int sum) {
        if (sum == k) {
            result.add(new ArrayList<>(subset));
            return;
        }
        if(sum > k) return;

        for (int i = start; i< size; i++) {
            subset.add(nums[i]);
            backtrack(nums,i, subset,sum+nums[i]);
            subset.remove(subset.size() - 1);
        }
    }

}

File: Data Structures & Algorithms/two-integer-sum/submission-0.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
         val sum = mutableMapOf<Int, Int>()
         for (i in 0..nums.size-1) {
            val dif = target - nums[i]
            if (sum[nums[i]] != null) {
                return intArrayOf(sum[nums[i]]!!, i)
            } else {
                sum[dif] = i
            }
         }
         return intArrayOf()
    }
}